component/prometheus/remotewrite/remote_write.go

```diff
@@ -72,6 +72,15 @@ func New(o component.Options, c Arguments) (*Component, error) {
 	oldDataPath := filepath.Join(o.DataPath, "wal", o.ID)
 	_ = os.RemoveAll(oldDataPath)
 
+	// When experimental delete WAL on startup is enabled, we delete the WAL. This is a useful workaround for a range
+	// of issues where processing WAL leads to a failure of the agent. This should be used sparingly and the root cause
+	// of each issue should be investigated.
+	if os.Getenv("EXPERIMENTAL_DELETE_WAL_ON_STARTUP") == "true" {
```

Comment:
Shouldn't this be a config option in the component config?

Reply:
I think a knob like this is a kind of an insurance policy that may come very useful one day (or even today), but in the ideal world it would not have to exist. So I would prefer this to be a special thing, perhaps even not documented. But I'm open to hear others' opinions and change this approach if needed.